pkg/gatherers/clusterconfig/pod_disruption_budgets_test.go

```diff
@@ -5,55 +5,114 @@ import (
+			pdbsToNamespace:   nil,
+			minAvailableToPdb: nil,
+			expCount:          0,
+		},
+		{
+			name: "one openshift PDB",
+			pdbsToNamespace: map[string]string{
+				"openshift-test": "testT",
+			},
+			minAvailableToPdb: map[string]int{
+				"testT": 1,
+			},
+			expCount: 1,
+		},
+		{
+			name: "one random PDB",
+			pdbsToNamespace: map[string]string{
+				"random": "testR",
+			},
+			minAvailableToPdb: map[string]int{
+				"testR": 1,
+			},
+			expCount: 0,
+		},
+		{
+			name: "one openshift, one random PDB",
+			pdbsToNamespace: map[string]string{
+				"openshift-test": "testT",
+				"random":         "testR",
+			},
+			minAvailableToPdb: map[string]int{
+				"testT": 1,
+				"testR": 1,
+			},
+			expCount: 1,
+		},
+		{
+			name: "multiple openshift PDBs",
+			pdbsToNamespace: map[string]string{
+				"openshift-test":    "testT",
+				"openshift-default": "testD",
+			},
+			minAvailableToPdb: map[string]int{
+				"testT": 1,
+				"testD": 2,
+			},
+			expCount: 2,
+		},
+	}
 
-	fakeNamespace := "fake-namespace"
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			coreClient := kubefake.NewSimpleClientset()
 
-	// name -> MinAvailabel
-	fakePDBs := map[string]string{
-		"pdb-four":  "4",
-		"pdb-eight": "8",
-		"pdb-ten":   "10",
-	}
-	for name, minAvailable := range fakePDBs {
-		_, err := coreClient.PolicyV1().
-			PodDisruptionBudgets(fakeNamespace).
-			Create(context.Background(), &policyv1.PodDisruptionBudget{
-				ObjectMeta: metav1.ObjectMeta{
-					Namespace: fakeNamespace,
-					Name:      name,
-				},
-				Spec: policyv1.PodDisruptionBudgetSpec{
-					MinAvailable: &intstr.IntOrString{StrVal: minAvailable},
-				},
-			}, metav1.CreateOptions{})
-		if err != nil {
-			t.Fatalf("unable to create fake pdbs: %v", err)
-		}
-	}
-	ctx := context.Background()
-	records, errs := gatherPodDisruptionBudgets(ctx, coreClient.PolicyV1())
-	if len(errs) > 0 {
-		t.Errorf("unexpected errors: %#v", errs)
-		return
-	}
-	if len(records) != len(fakePDBs) {
-		t.Fatalf("unexpected number of records gathered: %d (expected %d)", len(records), len(fakePDBs))
-	}
-	pdba, ok := records[0].Item.(record.ResourceMarshaller).Resource.(*policyv1.PodDisruptionBudget)
-	if !ok {
-		t.Fatal("pdb item has invalid type")
-	}
-	name := pdba.ObjectMeta.Name
-	minAvailable := pdba.Spec.MinAvailable.StrVal
-	if pdba.Spec.MinAvailable.StrVal != fakePDBs[name] {
-		t.Fatalf("pdb item has mismatched MinAvailable value, %q != %q", fakePDBs[name], minAvailable)
+			for namespace, name := range test.pdbsToNamespace {
+				_, err := coreClient.PolicyV1().
+					PodDisruptionBudgets(namespace).
+					Create(context.Background(), &policyv1.PodDisruptionBudget{
+						ObjectMeta: metav1.ObjectMeta{
+							Namespace: namespace,
+							Name:      name,
+						},
+						Spec: policyv1.PodDisruptionBudgetSpec{
+							MinAvailable: &intstr.IntOrString{IntVal: int32(test.minAvailableToPdb[name])},
```

Comment:
Is this required?

Reply:
The old unit test tested if the min. available was asserted to the correct PDB. I left it in there just in case, but I don't think gatherer can mismatch the pdb and its spec (MinAvailable).

Reply:
Ahh OK.